pkg/signing/signing.go

```diff
@@ -274,6 +275,51 @@ func getSigsPrimPart(fimg *sif.FileImage) (sigs []*sif.Descriptor, descr []*sif.
 	return
 }
 
+func getSigsAllPart(fimg *sif.FileImage) ([]*sif.Descriptor, []*sif.Descriptor, error) {
+	var dpart []*sif.Descriptor
+	var sigs []*sif.Descriptor
+	var err error
+
+	// Ensure theres a primary partition.
+	_, _, err = fimg.GetPartPrimSys()
+	if err != nil {
+		return nil, nil, fmt.Errorf("no primary partition found")
+	}
+
+	// Loop thrugh all the partitions, (skipping DataSignatures)
+	// and collect all the signatures for a data partition.
+	for _, d := range fimg.DescrArr {
+		if !d.Used {
+			continue
+		}
+		// No need to verify a signature.
+		if d.Datatype == sif.DataSignature {
+			continue
+		}
+		s, _, err := fimg.GetLinkedDescrsByType(d.ID, sif.DataSignature)
+		if err != nil {
+			// If a partition is not signed, print a warning.
+			// TODO: how will/should this efect the exit code?
+			sylog.Warningf("No signature found for %s\n", datatypeStr(d.Datatype))
+			continue
+		}
+
+		// For every signature part, there will be a data partition.
+		for _, sp := range s {
```

Comment:
I think this data duplication is ok for now since the need to do so should be eliminated with upcoming refactoring of this package

Reply:
Yes, this should be refactored later. But this will create a Descriptor, for every DataSignature, so if a partition is signed twice, there will be two Descriptors in that slice/array.